c_secure_comm.c

```diff
@@ -853,10 +856,12 @@ void add_session_key_to_list(session_key_t *s_key,
             "key.");
         existing_s_key_list->num_key = MAX_SESSION_KEY;
     }
-    copy_session_key(&existing_s_key_list->s_key[existing_s_key_list->rear_idx],
-                     s_key);
+    int index = existing_s_key_list->rear_idx;
+    copy_session_key(&existing_s_key_list->s_key[index], s_key);
     existing_s_key_list->rear_idx =
         (existing_s_key_list->rear_idx + 1) % MAX_SESSION_KEY;
+
+    return index;
```

Comment:
Add some error checking, and return the index where the new session key got added to.